provider/resource_extension.go

```diff
@@ -0,0 +1,118 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/observiq/bindplane-op-enterprise/model"
+	"github.com/observiq/terraform-provider-bindplane/client"
+	"github.com/observiq/terraform-provider-bindplane/internal/parameter"
+	"github.com/observiq/terraform-provider-bindplane/internal/resource"
+)
+
+func resourceExtension() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceExtensionCreate,
+		Update: resourceExtensionCreate,
+		Read:   resourceExtensionRead,
+		Delete: resourceExtensionDelete,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the extension.",
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    false,
+				Description: "The extension type to use for extension creation.",
+			},
+			"parameters_json": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "A JSON object with options used to configure the extension.",
+			},
+			"rollout": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				ForceNew:    false,
+				Description: "Whether or not to trigger a rollout automatically when a configuration is updated. When set to true, BindPlane OP will automatically roll out the configuration change to managed agents.",
+			},
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(maxTimeout),
+			Read:   schema.DefaultTimeout(maxTimeout),
+			Delete: schema.DefaultTimeout(maxTimeout),
+		},
+	}
+}
+
+func resourceExtensionCreate(d *schema.ResourceData, meta any) error {
+	bindplane := meta.(*client.BindPlane)
+
+	destType := d.Get("type").(string)
+	name := d.Get("name").(string)
+	rollout := d.Get("rollout").(bool)
```

Comment:
Keep in mind that this type assertion is safe and expected (as painful as it is). The provider framework guarantees that these options exist and are the right type.